exercises/sublist/src/Sublist.hs

```diff
@@ -1,3 +1,6 @@
 module Sublist (Sublist(..), sublist) where
 
+data Sublist = Equal | Sublist | Superlist | Unequal deriving (Eq, Show)
```

Comment:
Potentially controversial. If we don't do this though, we have to `DONT-TEST-STUB` Sublist. I think it's reasonable to do it because it's not the student really has a choice on how to do this, they'll just read the test file and do it this way.

compare to:

* `go-counting` where we define `data Color = Black | White`
* `connect` where we define `data Mark = Cross | Nought`
* `bowling` where we define `data BowlingError = IncompleteGame | InvalidRoll { rollIndex :: Int, rollValue :: Int }`
* `robot-simulator` where we define `data Bearing = North | East | South | West`